examples/ul2/src/super_glue/data.py

```diff
@@ -0,0 +1,262 @@
+        task,
+        split=split,
+        download_config=download_config,
+    )
+
+    log.info(
+        f'Starting tokenization by preprocessing over {num_workers} threads!')
+    prefix_column_order = _task_prefix_column_names[task]
+    target_extractor_map = _task_target_maps[task]
+    if isinstance(target_extractor_map, dict):
+        target_extractor = lambda example: target_extractor_map[example['label']
+                                                               ]
+    else:
+        assert callable(target_extractor_map)
+        target_extractor = target_extractor_map
+
+    def convert_padding_token(input_ids: List,
+                              attention_mask: List,
+                              new_pad_id: int = -100):
+        n_non_padded = sum(attention_mask)
+        n_total = len(input_ids)
+
+        if n_non_padded == n_total:
+            return input_ids
+
+        non_padded_input_ids = input_ids[:n_non_padded]
+        return non_padded_input_ids + ([new_pad_id] * (n_total - n_non_padded))
+
+    def tokenize_function(inp: Mapping[str, Any]):
+        # truncates sentences to max_length or pads them to max_length
+        if extra_prefix is not None:
+            prefix = f'{extra_prefix} {task}'
+        else:
+            prefix = task
+        text = [prefix]
+        for prefix_column in prefix_column_order:
+            text.append(f'{prefix_column}: {inp[prefix_column]}')
+        encoder_text = '\n'.join(text)
+
+        decoder_text = target_extractor(inp)
+
+        if decoder_only_format:
+            raise NotImplementedError  # TODO(Alex): Implement this!
+
+        else:
+            encoder_dict = tokenizer(
+                text=encoder_text,
+                padding='max_length',
+                max_length=max_seq_length,
+                truncation=True,
+            )
+
+            decoder_dict = tokenizer(
+                text=decoder_text,
+                padding='max_length',
+                max_length=20,
+                truncation=True,
+            )
+
+            # HF hardcodes the ignore_index to -100 in the loss, so ensure that
+            # the padding id is -100 for the loss-generating batch keys
+            decoder_dict['input_ids'] = convert_padding_token(
+                decoder_dict['input_ids'],
+                decoder_dict['attention_mask'],
+                new_pad_id=-100)
+
+            return {
+                'input_ids': encoder_dict.input_ids,
+                'attention_mask': encoder_dict.attention_mask,
+                'labels': decoder_dict.input_ids,
+                'decoder_attention_mask': decoder_dict.attention_mask,
+            }
+
+    assert isinstance(dataset, datasets.Dataset)
+    columns_to_remove = list(dataset[0].keys())
+    dataset = dataset.map(
+        tokenize_function,
+        batched=False,
+        num_proc=None if num_workers == 0 else num_workers,
+        # batch_size=3, #1000,
+        remove_columns=columns_to_remove,
```

Comment:
just flagging this so we don't miss it
```suggestion
        num_proc=None if num_workers == 0 else num_workers,
        remove_columns=columns_to_remove,
```